### DISCUSSION TITLE
Combo Data Merge

### DISCUSSION THREAD

**Discussion by LunaUrsa** [TripSit/drugs#22] [General]
In my work, I've come across an issue: there are two separate combo.json files.

First, we have what I'll call V1.
This is [what we have in the repo right now](https://github.com/TripSit/drugs/blob/main/combos.json). and looks like this:

```
[key: string] {
    [key: string]: {
        status: string;
        note?: string;
    }
}
```

Secondly we have V2
This is [the previous version of the same file](https://github.com/TripSit/drugs/blob/877b1cb59b1c6f9e1af954e7273bebde7d4fe44d/combos.json) and looks like:
```
[key: string] {
    [key: string]:  string;
}
```

Differences include:
V1 includes the "note" field.
V2 includes the "Serotonin Syndrome" status.
V2 uses "Safe & Decrease"     instead of "Low Risk & Decrease"
V2 uses "Safe & Synergy"       instead of "Low Risk & Synergy"
V2 uses "Safe & No Synergy" instead of "Low Risk & No Synergy"

V1 is being used by https://combo.tripsit.me/ and (I believe) the combo generator (since the combo chart does not include "serotonin syndrome").
Note that MDMA + Ampheatime = Low Risk & Synergy here.

~~V2 is being used by the API, and by extension, TripBot on Discord: https://tripbot.tripsit.me/api/tripsit/getInteraction/mdma/amphetamines.
Note that MDMA + Amphetamines = Unsafe here.~~
The API now uses V1.

I made a script that would compare the combo data in each file, and unfortunately, [nearly every combo has been changed](https://github.com/TripSit/drugs/pull/21/files), so it isn't possible to just do a quick merge of a few changes; **this seems like it's an all or nothing change.** 

~~ I find that V2's definitions are better. For example, V2 is already more in alignment with #19 by labeling "MDMA + Amphetamine = Unsafe," and I like the addition of "Serotonin Syndrome" instead of just "Dangerous." However, something in my mind tells me we intentionally moved away from labeling things as "safe," so the "low risk" is a better term. ~~ 
After discussion i dont agree with this anymore 

Thoughts?

**Comment by utaninja** [comment]
There are a series of issues. From my understanding-
1. There are inconsistent interactions. V1 has MDMA + Amphetamine as Low Risk + Synergy, and V2 MDMA + Amphetamines = Unsafe. These need to be reviewed.
1b. V1 has interaction explanations.

It seems from skimming various interactions, V1 seems to be most updated version.

Why does the API use V2? Was it not updated?
Response from MoonBear: https://gyazo.com/a1fbdfd26160dad3f88f814fecd67ca6

I do not think we should use serotonin syndrome. For a few reasons, one of these being that saying serotonin syndrome will hide it from other risks in the event there are. As in, even if there is a serotonin syndrome risk, there are also going to possibly be other interactions for a combination. If there is an individual that is willing to take the risk of serotonin syndrome (especially to note it is not always instadeath per say, and not all ss risk combos are a risk at X dosage) they will not be aware of other combination risks. Another reason is it is likely to put doubt onto the chart, a good reason to use caution is caution does not mean always dangerous. There is going to need to be a balance of what the risks we are warning people about, and I think caution or dangerous is better used for any combination that at least I can think of.

The words deadly and safe we should not use because they are in a sense full-stop. It is safe. We should not say safe, we should say safer. We should not say deadly, we should say has risks of being deadly.

Honestly, it seems we have a few different options.
1. Review every combination currently, considering the views of both versions, and making a suggestion on what it should be.
2. Read over the chart, and over time people will suggest changes if found. Such as https://github.com/TripSit/drugs/pull/19. While yes, there may be things on V2 that is more "correct" we may be able to posit that these kinds of "wrong" interactions will be found and adjusted before the new version is released.

**Reply by LunaUrsa** [reply]
> Why does the API use V2? Was it not updated?

The API uses the files i could find that got the API to work again. Remember that this is a decade old project and i didn't build this, so i just got it working and stumbled upon the other file later. We're trying to do the update now

I generally agree with not using SS now, i can see your logic there, and considering the official chart uses "low risk" instead of "safe" it does seem like using V1 would be the better option and then we can review and update that one as we go.

The API and tripbot now uses V1 for its combo info

**Reply by utaninja** [reply]
So we can note that it may be beneficial to review V2 in general if we want another insight for an opinion, but this should be resolved then?

**Reply by LunaUrsa** [reply]
I'd like to do another comparison where we see if any changes were made that weren't just renaming, but I think we're pretty safe to use V1 for now
